packages/tools/kadena-cli/src/tx/commands/txStatus.ts

```diff
@@ -10,33 +10,59 @@ import { globalOptions } from '../../utils/globalOptions.js';
 import { log } from '../../utils/logger.js';
 import { txOptions } from '../txOptions.js';
 
-export const getTxStatus = async (config: {
+export const getTxStatus = async ({
+  requestKey,
+  chainId,
+  networkConfig,
+  txPoll,
+}: {
   requestKey: string;
   chainId: ChainId;
-  networkConfig: INetworkCreateOptions;
+  networkConfig: Omit<INetworkCreateOptions, 'networkExplorerUrl'>;
+  txPoll: boolean;
 }): Promise<CommandResult<ICommandResult>> => {
+  const notFoundErrorMessage = `No Transaction found for requestkey "${requestKey}" on network "${networkConfig.networkId}" and chain "${chainId}".`;
   try {
-    const { pollStatus } = createClient(
-      `${config.networkConfig.networkHost}/chainweb/0.0/${config.networkConfig.networkId}/chain/${config.chainId}/pact`,
-    );
-    const result = await pollStatus(
-      {
-        requestKey: config.requestKey,
-        chainId: config.chainId,
-        networkId: config.networkConfig.networkId,
-      },
-      { timeout: 25000 },
+    const { getStatus, pollStatus } = createClient(
+      `${networkConfig.networkHost}/chainweb/0.0/${networkConfig.networkId}/chain/${chainId}/pact`,
     );
+    let result = null;
+    const payload = {
+      requestKey: requestKey,
+      chainId: chainId,
+      networkId: networkConfig.networkId,
+    };
+    if (txPoll) {
+      result = await pollStatus(payload, {
+        timeout: 60000,
+      });
+    } else {
+      result = await getStatus(payload);
+    }
+
+    const trimmedRequestKey = requestKey.endsWith('=')
```

Comment:
why is this trimming here?

Reply:
I'm not familiar with this behaviour, it seems odd. Is it documented anywhere? would be very helpful to have a commented link or something for future maintenance to make it clear why this is here.